os/arch/arm/src/amebasmart/amebasmart_idle.c

```diff
@@ -169,6 +159,10 @@ void up_idle(void)
 
 	nxsched_process_timer();
 #else
+	/* power saving when idle*/
+	__asm(" DSB");
+	__asm(" WFI");
+	__asm(" ISB");
 	/* set core to WFE */
 	__asm("WFE");
```

Comment:
Could you check it?
WFI is already call, but after get irq, WFE will be called again.

Reply:
And Please check if it is correct to call WFI here.

This location can be called even when not in pm_sleep state, so in a normal situation, even though there are events on the secondary core, the primary core may not wake up.

Reply:
@ewoodev I asked @edwakuwaku, he will do necessary cleanup required for SMP in PR #6256

Reply:
Thank you,
when we use single core only, we can use WFI instead WFE.
WFI can reduce power more.

Reply:
Yes WFI is better than WFE